src/DisplayPlug.vala

```diff
@@ -46,13 +49,38 @@ public class Display.Plug : Switchboard.Plug {
             displays_view = new DisplaysView ();
             var action_bar = new Gtk.ActionBar ();
             action_bar.get_style_context ().add_class (Gtk.STYLE_CLASS_INLINE_TOOLBAR);
+            
+            detect_touchscreen ();
+
+            var schema_source = GLib.SettingsSchemaSource.get_default ();
+            var rotation_lock_schema = schema_source.lookup ("org.gnome.settings-daemon.peripherals.touchscreen", true);
+            if (rotation_lock_schema == null) {
+	            info ("Schema \"org.gnome.settings-daemon.peripherals.touchscreen\" is not installed on your system.");
+	            rotation_lock_setting = (GLib.Settings) null;
+            } else {
+	            rotation_lock_setting = new GLib.Settings.full (rotation_lock_schema, null, null);
+            }
 
             var mirror_grid = new Gtk.Grid ();
             mirror_grid.margin = 12;
             mirror_grid.column_spacing = 6;
             mirror_grid.orientation = Gtk.Orientation.HORIZONTAL;
             var mirror_label = new Gtk.Label (_("Mirror Display:"));
             var mirror_switch = new Gtk.Switch ();
+            mirror_grid.add (mirror_label);
+            mirror_grid.add (mirror_switch);
+
+            var rotation_lock_grid = new Gtk.Grid ();
+            rotation_lock_grid.margin = 12;
+            rotation_lock_grid.column_spacing = 6;
+            rotation_lock_grid.orientation = Gtk.Orientation.HORIZONTAL;
+            var rotation_lock_label = new Gtk.Label (_("Rotation Lock:"));
+            var rotation_lock_switch = new Gtk.Switch ();
+            rotation_lock_switch.active = rotation_lock_setting.get_boolean ("orientation-lock");
+            if (has_touchscreen && rotation_lock_schema != null) {
```

Comment:
This `if` statement could be moved to where the lock grid construction happens. There is no need to construct a new grid and it's children when there is no touchscreen schema installed.

Another thing worth a look is to simplify this check. If you apply my previous suggestion about creating the schema only when the system has a touchscreen device, this statement can be reduced to only:
```
if (rotation_lock_schema != null) {

}
```